### ISSUE TITLE
Different answers for Chapter 2 Exercises 1a and 1b 

### ISSUE THREAD

**Issue by ghost** [onmee/ISLR-Answers#1]
Hey Onmee, I got the following for Questions 1a) and 1b)

Conceptual 
1. For each of parts (a) through (d), indicate whether we would generally expect the performance of a flexible statistical learning method to be better or worse than an inflexible method. Justify your answer.
(a) The sample size n is extremely large, and the number of predic¬ tors p is small.
With large sample sizes, a flexible learning method is likely to vary more across training data, which increases variance. 
(b) The number of predictors p is extremely large, and the number of observations n is small.
With small sample sizes, a flexible learning method is less likely to read into noise, which decreases variance.

You know more than me so if you'd be willing I'd like to hear your justification for your answers?

**Comment by ghost** [comment]
Updated my answers, to me this makes more sense now :)


>  (a) The sample size n is extremely large, and the number of predic¬ tors p is small.
> WRONG: With large sample sizes, a flexible learning method is likely to vary more across training data, which increases variance. 
> CORRECT: With large sample sizes, a flexible learning method will have more observations to guide its fit, which makes it less likely to read into errors and outliers.
> 
> (b) The number of predictors p is extremely large, and the number of observations n is small.
> WRONG: With small sample sizes, a flexible learning method is less likely to read into noise, which decreases variance.
> CORRECT: With small sample sizes, a flexible learning method has less observations to guide its fit and could likely read into noise. Imagine three points where the middle point  is towards the bottom of the graph (perhaps owing to a measurement error) while the rest are towards the top. This outlier would affect the non-flexible model through a decrease in slope, while a flexible model would probably dip down to the outlier and then back up and thus fit the error.

**Comment by onmee** [comment]
Hi,

I'm glad that you are finding my answers somewhat useful, and as I mentioned in the Readme, always confirm it yourself. I did this a while ago so don't remember exactly, but in general the reasons are:

A higher flexibility in effect means more parameters/features are being used as predictors. In other words the model has higher dimensionality or degress of freedom . This means the more flexible a model is, the more accurately it will fit the training data, which in general will lead to overfitting. The MSE (mean squared error) of the model on the training set will reduce consistently as you increase the flexibility. Whereas on the test set, the MSE will reduce to a minimum at some level of flexibility and then increase, as it starts to overfit.

So, for (a), your reasoning is correct. Larger sample sizes necessarily mean a better representation of the whatever is being sampled, so in general lower margine for errors, lower standard deviation etc. And a flexible model with a smaller number of predictors in general will hit the optimal spot for low MSE than an inflexible method. 

Your reasoning for (b) is correct too. It would almost certainly overfit because the flexibility would be very large, which is exacerbated by a small dataset - where it would likely fit the curve through outliers - leading to lower training MSE and higher test MSE.

**Comment by ghost** [comment]
Your introduction of training vs test data into this discussion is appreciated. On your last point though, that a flexible method on a small dataset would lead to lower training MSE (agree) but higher test MSE -- what's the intuition behind the higher test MSE part? You're free to respond or not to respond, I'll figure it out eventually. Thanks.
